internal/services/containers/kubernetes_fleet_manager_resource.go

```diff
@@ -121,40 +144,51 @@ func (r KubernetesFleetManagerResource) Create() sdk.ResourceFunc {
 	}
 }
 
-func (r KubernetesFleetManagerResource) Read() sdk.ResourceFunc {
+func (r KubernetesFleetManagerResource) Update() sdk.ResourceFunc {
 	return sdk.ResourceFunc{
-		Timeout: 5 * time.Minute,
+		Timeout: 30 * time.Minute,
 		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
 			client := metadata.Client.ContainerService.V20231015.Fleets
-			schema := KubernetesFleetManagerResourceSchema{}
 
 			id, err := fleets.ParseFleetID(metadata.ResourceData.Id())
 			if err != nil {
 				return err
 			}
 
+			var model KubernetesFleetManagerModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
 			resp, err := client.Get(ctx, *id)
 			if err != nil {
-				if response.WasNotFound(resp.HttpResponse) {
-					return metadata.MarkAsGone(*id)
-				}
 				return fmt.Errorf("retrieving %s: %+v", *id, err)
 			}
 
-			if model := resp.Model; model != nil {
-				schema.Name = id.FleetName
-				schema.ResourceGroupName = id.ResourceGroupName
-				r.mapFleetToKubernetesFleetManagerResourceSchema(*model, &schema)
+			properties := resp.Model
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", *id)
+			}
+			if resp.Model.Properties == nil {
+				return fmt.Errorf("retrieving %s: `properties` was nil", *id)
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				properties.Tags = pointer.To(model.Tags)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *properties, fleets.DefaultCreateOrUpdateOperationOptions()); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
```

Comment:
We're not patching anything into properties so there's no need to nil check this.
```suggestion
			if resp.Model == nil {
				return fmt.Errorf("retrieving %s: `model` was nil", *id)
			}
			
			payload := resp.Model

			if metadata.ResourceData.HasChange("tags") {
				payload.Tags = pointer.To(model.Tags)
			}

			if err := client.CreateOrUpdateThenPoll(ctx, *id, *payload, fleets.DefaultCreateOrUpdateOperationOptions()); err != nil {
				return fmt.Errorf("updating %s: %+v", *id, err)
```